overlord/auth/auth_test.go

```diff
@@ -312,15 +322,31 @@ func (as *authSuite) TestGetAuthenticatorFromUser(c *C) {
 }
 
 func (as *authSuite) TestAuthenticatorSetHeaders(c *C) {
+	root, err := as.makeTestMacaroon()
+	c.Check(err, IsNil)
+	discharge, err := as.makeTestDischarge()
+	c.Check(err, IsNil)
+
+	serializedMacaroon, err := auth.MacaroonSerialize(root)
+	c.Check(err, IsNil)
+	serializedDischarge, err := auth.MacaroonSerialize(discharge)
+	c.Check(err, IsNil)
+
 	as.state.Lock()
-	user, err := auth.NewUser(as.state, "username", "macaroon", []string{"discharge"})
+	user, err := auth.NewUser(as.state, "username", serializedMacaroon, []string{serializedDischarge})
 	as.state.Unlock()
 	c.Check(err, IsNil)
 
 	req, _ := http.NewRequest("GET", "http://example.com", nil)
 	authenticator := user.Authenticator()
 	authenticator.Authenticate(req)
 
+	// discharge macaroon should be bound to the root macaroon
+	discharge.Bind(root.Signature())
+	serializedPreparedDischarge, err := auth.MacaroonSerialize(discharge)
+	c.Check(err, IsNil)
+
 	authorization := req.Header.Get("Authorization")
-	c.Check(authorization, Equals, `Macaroon root="macaroon", discharge="discharge"`)
+	expected := fmt.Sprintf(`Macaroon root="%s", discharge="%s"`, serializedMacaroon, serializedPreparedDischarge)
```

Comment:
I see your point. This particular test goal is to make sure the Authorization header is set after Authenticate(req) call, using the user macaroons (and expecting the discharge to be prepared). Should I pre-calculate the expected header?
FWIW, there is some test verifying our MacaroonDeserialize can deserialize a macaroon from the store that I'm updating to also confirm that re-serializing the macaroon gives us the same original store serialization.